### ISSUE TITLE
Add more user-friendly warnings on dependencies error

### ISSUE THREAD

**Issue by Megamiun** [atom-haskell/haskell-ghc-mod#225]
Hello, I had the same problem as #223 today. Though I did manage to correct it(installing ghc-mod) after reading the message, I think it would be interesting to have a more user-friendly message, without a stacktrace and explaining to the user the problem and some simple solutions.

Did something similar on the project atom-python-run and I really believe it helped a little with the user experience. If you want, I can try to do it, just give me a heads-up if it is desired.

**Comment by lierdakil** [comment]
The whole dance around stacktraces and whatnot started initially as a way to

1. Give users an easy way to get support, which would automatically include all relevant information.
2. Get some feedback on what *can* go wrong in general

A caveat with that approach is that message is rather scary-looking (well, Atom default, can't do much about that), and a kneejerk reaction might be to click on "create issue" without reading. Doing something about that *has* been in the back of my mind for a while, but I didn't get to implementation yet.

Help is welcome. Bear in mind, there are two types of errors that tend to happen wrt ghc-mod executable and which probably *need* more user-friendly messages:

- `ENOENT`, which means that either there is no 'ghc-mod' executable installed or Atom can't find it for whatever reason
- `EACCESS`, which, in most cases, indicates configuration error

Also should probably "remember" if ghc-mod is built with a wrong compiler version, and in case of problems re-report that. But that's a different story, and builder selection (i.e. choice between cabal, stack or neither) must happen first.

A good idea is to check for typical configuration errors:

- 'Additional Path Directories' contains non-directories
- 'Ghc Mod Path' contains a directory (usual culprit for `EACCESS`)

**Comment by lierdakil** [comment]
Closing in favor of #101
